### PR TITLE
Better and more colorful boot verbose

### PR DESCRIPTION
**Pull request by pedrofracassi**
image

- [x] Make APIs a CommandRequirement and log when a command doesn't load because one is missing
- [x] Make environment variables a CommandRequirement and log when a command doesn't load because one is missing
- [x] Make environment variables a requirement (how?) for APIs and log when a command doesn't load because one is missing
- [x] Let `canLoad` return a String and use it as the error text
- [x] Check for all three required environment variables for Lavalink instead of just `LAVALINK_HOST`

### CODE REVIEW

**Comment by pedrofracassi** [comment]
image

Ready for review!

**Comment by pedrofracassi** [comment]
**Please wait for all reviewers before merging. This one has the potential to break some stuff.**

File: src/listeners/MainListener.js [hunk 1]
```diff
@@ -22,9 +22,9 @@ module.exports = class MainListener extends EventListener {
         user: this.user.id,
         shards: 1
       })
-      this.log('Player manager connection established!', 'Music')
+      this.log('[92mLavalink connection established!', 'Music')
     } else {
-      this.log('Player manager connection didn\'t establish!', 'Music')
+      this.log('[91mLavalink connection didn\'t establish - LAVALINK_HOST not set.', 'Music')
```
Comment (perronosaurio) [inline src/listeners/MainListener.js]: What if I have LAVALINK_HOST set and the error here is another? :thonk:

Comment (pedrofracassi) [reply src/listeners/MainListener.js]: Tru dat. We should check for all Lavalink variables instead.

File: src/Switchblade.js [hunk 2]
```diff
@@ -180,16 +180,23 @@ module.exports = class Switchblade extends Client {
...
+      return false
     }
+
+    if (api.canLoad() !== true) {
+      this.log(`[91m${api.name} failed to load - ${api.canLoad() || 'canLoad function did not return true.'}`, 'APIs')
+      return false
+    }
+
+    if (!api.envVars.every(variable => {
+      if (!process.env[variable]) this.log(`[91m${api.name} failed to load - Required environment variable "${variable}" not set.`, 'APIs')
```
Comment (Lireoy) [inline src/Switchblade.js]: Small grammatical improvement:
Required environment variable "${variable}" `is` not set.

File: src/listeners/MainListener.js [hunk 3]
```diff
@@ -24,7 +25,7 @@ module.exports = class MainListener extends EventListener {
       })
       this.log('[92mLavalink connection established!', 'Music')
     } else {
-      this.log('[91mLavalink connection didn\'t establish - LAVALINK_HOST not set.', 'Music')
+      this.log(`[91mLavalink connection didn\'t establish - Rrequired environment variable(s) (${lavalinkRequiredVariables.filter(variable => !process.env[variable]).join(', ')}) not set.`, 'Music')
```
Comment (Lireoy) [inline src/listeners/MainListener.js]: Possible improvement idea:
`Failed to establish Lavalink connection`

Comment (Doges) [reply src/listeners/MainListener.js]: outdated

**Comment by pedrofracassi** [comment]
I love how Pill completely ignores other people's reviews and just merges it

**Comment by bolsomito** [comment]
who


cares

**Comment by pedrofracassi** [comment]
Me, who once had the idea to make a bot with multiple people to keep the code quality good?

**Comment by Doges** [comment]
i want to do some new things on it

**Pull request merged**